prov/usnic/src/usdf_msg.c

```diff
@@ -1096,6 +1096,7 @@ usdf_msg_handle_recv(struct usdf_domain *udp, struct usd_completion *comp)
 		*/
 		if (rxlen > 0) {
 			rqe->ms_length -= rxlen;
+			comp->uc_status = USD_COMPSTAT_ERROR_TRUNC;
```

Comment:
I think I'd rather see a `usdf_msg_recv_complete_err` equivalent of `usdf_msg_recv_complete`.  That function would then be responsible for propagating the error back up to the soft CQ instead of using `comp` as a partial in/out parameter.